doc/source/developer_overview.rst

```diff
@@ -210,115 +213,123 @@ For a complete definition of the ``rcl`` |API|, see the |API| documentation:
+The ``rosidl`` repository, located on |GitHub|_ at `ros2/rosidl <https://github.com/ros2/rosidl>`_, defines the message IDL syntax, i.e. syntax of ``.msg`` files, ``.srv`` files, etc., and contains |packages| for parsing the files, for providing CMake infrastructure to generate code from the messages, for generating implementation agnostic code (headers and source files), and for establishing the default set of generators.
 The repository contains these |packages|:
 
--  ``rosidl_cmake``: provides CMake functions and Modules for generating code from ``rosidl`` files, e.g. ``.msg``, ``.srv``, etc.
--  ``rosidl_default_generators``: defines the list of default generators which ensures that they are installed as dependencies, but other injected generators will also be used.
+-  ``rosidl_cmake``: provides CMake functions and Modules for generating code from ``rosidl`` files, e.g. ``.msg`` files, ``.srv`` files, etc.
+-  ``rosidl_default_generators``: defines the list of default generators which ensures that they are installed as dependencies, but other injected generators can also be used.
 -  ``rosidl_generator_c``: provides tools to generate C header files (``.h``) for ``rosidl`` files.
 -  ``rosidl_generator_cpp``: provides tools to generate C++ header files (``.hpp``) for ``rosidl`` files.
+-  ``rosidl_generator_py``: provides tools to generate Python modules for ``rosidl`` files.
 -  ``rosidl_parser``: provides Python |API| for parsing ``rosidl`` files.
 
-Generators for other languages, e.g. ``rosidl_generator_py``, are hosted externally (in a different repository) but would use the same mechanism that ``rosidl_generator_c`` and ``rosidl_generator_cpp`` use to "register" themselves as a ``rosidl`` generator.
+Generators for other languages, e.g. ``rosidl_generator_java``, are hosted externally (in different repositories) but would use the same mechanism that the above generators use to "register" themselves as a ``rosidl`` generator.
+
+In addition to the aforementioned |packages| for parsing and generating headers for the ``rosidl`` files, the ``rosidl`` repository also contains |packages| concerned with "type support" for the message types defined in the files.
+Type support refers to the ability to interpret and manipulate the information represented by ROS message instances of particular types (serialization and deserialization, for example).
+Type support can either be provided by code that is generated at compile time or it can be done programmatically based on the contents of the ``rosidl`` file and the data received, by introspecting the data.
```

Comment:
I think it would help to clarify `rosidl file` with `(.msg and .srv files)`. Or otherwise we should consider making it a term in the glossary and linking it here.